Library/Homebrew/cask/lib/hbc/installer.rb

```diff
@@ -420,10 +437,38 @@ def zap
       purge_caskroom_path
     end
 
+    def backup_path
+      return nil if @cask.staged_path.nil?
+      Pathname.new "#{@cask.staged_path}.upgrading"
+    end
+
+    def backup_metadata_path
+      return nil if @cask.metadata_versioned_path.nil?
+      Pathname.new "#{@cask.metadata_versioned_path}.upgrading"
+    end
+
     def gain_permissions_remove(path)
       Utils.gain_permissions_remove(path, command: @command)
     end
 
+    def purge_backed_up_versioned_files
+      ohai "Purging files for version #{@cask.version} of Cask #{@cask}"
+
+      # versioned staged distribution
+      gain_permissions_remove(backup_path) if !backup_path.nil? && backup_path.exist?
+
+      # Homebrew-Cask metadata
+      if backup_metadata_path.respond_to?(:children) &&
+         backup_metadata_path.exist? &&
```

Comment:
The `&&` shouldn‘t be here.

Reply:
The whole condition here could probably just be `backup_metadata_path.directory?`.